velox/functions/sparksql/ArrayInsertFunction.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Udf.h"
+
+namespace facebook::velox::functions::sparksql {
+
+/// array_insert(array(E), pos, E, bool) → array(E)
+/// Places new element into index pos of the input array.
+template <typename T>
+struct ArrayInsertFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+
+  // INT_MAX - 15, keep the same limit with spark.
+  static constexpr int32_t kMaxNumberOfElements = 2147483632;
+
+  FOLLY_ALWAYS_INLINE bool callNullable(
+      out_type<Array<Generic<T1>>>& out,
+      const arg_type<Array<Generic<T1>>>* srcArray,
+      const arg_type<int32_t>* pos,
+      const arg_type<Generic<T1>>* item,
+      const arg_type<bool>* legacyNegativeIndex) {
+    VELOX_USER_CHECK_NOT_NULL(
+        legacyNegativeIndex,
+        "Parameter legacyNegativeIndex should not be NULL.")
```

Comment:
It seems in Spark legacyNegativeIndex is an optional variable which can be accessed through config. Shall we make it optional and add an initialize method like https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/Rand.h#L31 when it is not provided?

Reply:
Also cc @PHILO-HE, I saw you made some similar changes for `size` function. Can you please also take a look? Thanks.

Reply:
Either way is OK to me. Directly passing the config to native function may make code concise if this config is just used in one place.

Reply:
Makes sense to me. Thanks.